api/src/exam-environment/routes/exam-environment.test.ts

```diff
@@ -0,0 +1,601 @@
+          }
+        });
+
+        const body2: Static<typeof examEnvironmentPostExamGeneratedExam.body> =
+          {
+            examId: mock.examId
+          };
+
+        const res2 = await superPost('/exam-environment/exam/generated-exam')
+          .send(body2)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res2).toMatchObject({
+          status: 429,
+          body: {
+            code: 'FCC_EINVAL_EXAM_ENVIRONMENT_PREREQUISITES'
+          }
+        });
+
+        await fastifyTestInstance.prisma.envExamAttempt.update({
+          where: {
+            id: recentExamAttempt.id
+          },
+          data: {
+            // Set start time such that exam has expired, but 24 hours + 1s has passed
+            startTimeInMS:
+              Date.now() -
+              (mock.exam.config.totalTimeInMS + (24 * 60 * 60 * 1000 + 1000))
+          }
+        });
+        // Generate new exam for user to be assigned
+        const newGeneratedExam = structuredClone(mock.generatedExam);
+        newGeneratedExam.id = mock.oid();
+        await fastifyTestInstance.prisma.envGeneratedExam.create({
+          data: newGeneratedExam
+        });
+
+        const body3: Static<typeof examEnvironmentPostExamGeneratedExam.body> =
+          {
+            examId: mock.examId
+          };
+
+        const res3 = await superPost('/exam-environment/exam/generated-exam')
+          .send(body3)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        // Time is greater than 24 hours. So, request should pass, and new exam should be generated
+        expect(res3).toMatchObject({
+          status: 200,
+          body: {
+            data: {
+              examAttempt: {
+                // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+                id: expect.not.stringMatching(mock.examAttempt.id)
+              }
+            }
+          }
+        });
+      });
+
+      it('should return the current attempt if it is still ongoing', async () => {
+        const latestAttempt =
+          await fastifyTestInstance.prisma.envExamAttempt.create({
+            data: mock.examAttempt
+          });
+
+        const body: Static<typeof examEnvironmentPostExamGeneratedExam.body> = {
+          examId: mock.examId
+        };
+
+        const res = await superPost('/exam-environment/exam/generated-exam')
+          .send(body)
+          .set(
+            'exam-environment-authorization-token',
+            examEnvironmentAuthorizationToken
+          );
+
+        expect(res).toMatchObject({
+          status: 200,
+          body: {
+            data: {
+              examAttempt: latestAttempt
+            }
+          }
+        });
+      });
+
+      it('should return an error if Camper has taken all generated exams', async () => {
```

Comment:
```suggestion
      it('should return an error if the database has insufficient generated exams', async () => {
```
Bit of a nit, but it makes it clearer that _we_ are at fault in this case, not the camper.